### ISSUE TITLE
Hyperparameter for training

### ISSUE THREAD

**Issue by krantiparida** [swimmiing/ACL-SSL#2]
Hi,
Thanks for making the code public for this interesting work. 

I was trying to train the network on VGG. I noticed there is a difference between the batch size and learning rate mentioned in the paper compared to the one mentioned in the config file [here](https://github.com/swimmiing/ACL-SSL/blob/main/config/train/Exp_ACL_v1.yaml). The learning rate and batch size are mentioned in the paper to be `1e-3` and `16`, whereas in the config file, the values are `1e-4` and `8`. I tried training with both settings, I observed that with the values `(1e-3` and `16`)` mentioned in the paper, the training loss diverges whereas the values for the config file seem to perform well.  I am attaching the screenshot of both curves below for your reference.
![Screenshot 2024-03-28 at 11 28 03](https://github.com/swimmiing/ACL-SSL/assets/34489236/1d8ad504-39a4-48bb-b2f4-d9c3ca64536b)
![Screenshot 2024-03-28 at 11 28 20](https://github.com/swimmiing/ACL-SSL/assets/34489236/0f42eb7f-40e1-439e-a4db-9f49b98d7a09)

Also, in practice, the batch size for InfoNCE loss is usually higher (~128), is there any specific reason you have chosen a small batch size (8/16)? And what can be the reason for the curve diverging for a higher batch size?

Thanks.

**Comment by krantiparida** [comment]
Thanks for the prompt response. I will use the values mentioned in the configuration file.
Also, one clarification, when you say 8 per GPU, it means the InfoNCE loss is calculated per GPU by taking the batch size as 8 (8 class classification as the loss function) and then the gradients summed for both the GPUs or does it perform 16 class classification by combining predictions from both the gpus?

**Comment by swimmiing** [comment]
The first one you mentioned is correct. Using DDP, the infoNCE loss is calculated by batch size 8 in each GPU and then aggregating the gradients.

**Comment by krantiparida** [comment]
Thanks for the clarification.
